system-addon/lib/TopSitesFeed.jsm

```diff
@@ -137,8 +136,12 @@ this.TopSitesFeed = class TopSitesFeed {
       data: this._getPinnedWithData()
     }));
   }
-  onAction(action) {
+  async onAction(action) {
     switch (action.type) {
+      case at.INIT:
```

Comment:
And on the flip side, if tippyTop does resolve sooner than `NEW_TAB_LOAD`, the state will already be updated and not refreshed due to the 15-minute check.